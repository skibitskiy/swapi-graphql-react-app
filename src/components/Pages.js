import React, { memo } from 'react';
import { MenuList, MenuItem, Paper } from '@material-ui/core';

function Pages(props) {
    const { count, onClick } = props;

    return (
        <div style={{
            height: '100vh',
            position: 'fixed',
            display: 'flex',
            alignItems: 'center',
            left: '16px'
        }}>
            <Paper>
                <MenuList onClick={onClick}>
                    { Array.from(Array(count), ((_, i) => <MenuItem data-page={i + 1} key={i}>{ i + 1 }</MenuItem>)) }
                </MenuList>
            </Paper>
        </div>
    )
}

export default memo(Pages);
