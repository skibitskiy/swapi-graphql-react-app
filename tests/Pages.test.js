import React from 'react';
import * as test from '@testing-library/react';
import user from '@testing-library/user-event';
import Pages from '../src/components/Pages';

describe('Pages component', () => {
    it('Snapshot', () => {
        const mockedOnClick = jest.fn();
        const { container } = test.render(<Pages count={3} onClick={mockedOnClick}/>) ;
        expect(container).toMatchSnapshot();
    })

    it('onClick check', () => {
        const mockedOnClick = jest.fn();
        test.render(<Pages count={3} onClick={mockedOnClick}/>) ;

        ['1', '2'].forEach((text) => {
            const button = test.screen.getByText(text);
            user.click(button);
        });
        expect(mockedOnClick.mock.calls.length).toBe(2)
    })
});
