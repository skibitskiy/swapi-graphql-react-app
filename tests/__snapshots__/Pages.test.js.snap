// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages component Snapshot 1`] = `
<div>
  <div
    style="height: 100vh; position: fixed; display: flex; align-items: center; left: 16px;"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <ul
        class="MuiList-root MuiList-padding"
        role="menu"
        tabindex="-1"
      >
        <li
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
          data-page="1"
          role="menuitem"
          tabindex="0"
        >
          1
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
          data-page="2"
          role="menuitem"
          tabindex="-1"
        >
          2
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
          data-page="3"
          role="menuitem"
          tabindex="-1"
        >
          3
          <span
            class="MuiTouchRipple-root"
          />
        </li>
      </ul>
    </div>
  </div>
</div>
`;
